pkg/synopsysctl/cmd_migrate.go

```diff
@@ -324,22 +348,154 @@ func migrateBlackDuck(namespace string) error {
 		}
 
 		if blackDuck.Spec.PersistentStorage {
-			err = removePVC(blackDuckNamespace)
+			var rabbitmqRCName, zookeeperRCName, uploadCacheRCName, uploadCacheKeyPVCName, uploadCacheDataPVCName string
+			if value, ok := blackDuck.Annotations["synopsys.com/created.by"]; ok && "pre-2019.6.0" == value {
+				rabbitmqRCName = "rabbitmq"
+				zookeeperRCName = "zookeeper"
+				uploadCacheRCName = "uploadcache"
+				uploadCacheKeyPVCName = "blackduck-uploadcache-key"
+				uploadCacheDataPVCName = "blackduck-uploadcache-data"
+			} else {
+				rabbitmqRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "rabbitmq")
+				zookeeperRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper")
+				uploadCacheRCName = util.GetResourceName(blackDuckName, util.BlackDuckName, "uploadcache")
+				uploadCacheKeyPVCName = fmt.Sprintf("%s-blackduck-uploadcache-key", blackDuckName)
+				uploadCacheDataPVCName = fmt.Sprintf("%s-blackduck-uploadcache-data", blackDuckName)
+			}
+			// scale down zookeeper
+			err = scaleDownRC(blackDuckNamespace, zookeeperRCName)
+			if err != nil {
+				return err
+			}
+			// scale down upload cache
+			err = scaleDownRC(blackDuckNamespace, uploadCacheRCName)
 			if err != nil {
 				return err
 			}
+
+			isSourceCodeEnabled := isSourceCodeEnabled(blackDuck.Spec.Environs)
+			if isSourceCodeEnabled {
+				err = migrateUploadCachePVCJob(blackDuckNamespace, blackDuckName, uploadCacheKeyPVCName, uploadCacheDataPVCName)
+				if err != nil {
+					return err
+				}
+			}
+
+			pvcs := []string{"blackduck-rabbitmq"}
+			if value, ok := blackDuck.Annotations["synopsys.com/created.by"]; ok && "pre-2019.6.0" == value {
+				pvcs = append(pvcs, "zookeeper-data", "zookeeper-datalog")
+			} else {
+				pvcs = append(pvcs, util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper-data"), util.GetResourceName(blackDuckName, util.BlackDuckName, "zookeeper-datalog"))
+			}
+			for _, pvc := range pvcs {
+				// check for an existance of PVC
+				_, err := util.GetPVC(kubeClient, blackDuckNamespace, pvc)
+				if err == nil {
+					if "blackduck-rabbitmq" == pvc {
```

Comment:
Scaling down and removing PVC are really separate actions.  Suggest:
for _, pvc := range pvcs{
  If "blackduck-rabbitmq" == pvc {
  ...
  }
  if err := removePVC(...); err != nil {
    return err
  }

And have removePVC do a get and delete if no error (like before this change)

Reply:
@rrati correct but whenever there is PVC for rabbitmq (only for 2019.4.x operator case), then only scale down..